### DISCUSSION TITLE
Authentication failed

### DISCUSSION THREAD

**Discussion by FiCan** [fredemmott/StreamDeck-Discord#36] [Q&A] [answered]
Hi there. Hoping I can find some help. I've followed the installation instructions, but on connecting to Discord, I get a "AUTHENTICATION_FAILED" response.
Things I've done:
- Followed instructions
- Used exactly `https://localhost` as the redirect URL
- Removed all other authenticated apps in Discord (like Logitech GHub) to avoid conflicts
- Triple checked the app ID and secret are correct
- Restarted the Streamdeck app
- Restarted the Discord app
- Recreated the discord developer app following the instructions again

No joy unfortunetly. Discord is receiving the request and I'm approving it, but the Streamdeck app just shows auth failed every time in the debug info. Any idea of what else I could try or may be conflicting?

Cheers!

**Comment by fredemmott** [comment]
`https://localhost/` is required, including the trailing slash. Does that fix it for you?

**Comment by FiCan** [comment]
I did try that too, same result.

**Reply by lasmith05** [reply]
I had auth error briefly. Restarting discord and reauth/reconnecting worked for me. (It asked for permission and then permission to grab user name, etc)

**Reply by SwissStar** [reply]
Are you running Discord as admin? Had the same issue, startet Discord without admin and it worked again.

**Comment by jackblovett** [comment]
I had this issue as well, I realized I made the application on my alternate account for discord, in order for it to work you need to make the application on the same account you want to use it.

**Reply by fredemmott** [reply]
FYI you can use a single discord app for up to 50 discord accounts - add them on this page:

<img width="1256" alt="image" src="https://user-images.githubusercontent.com/360927/103572888-ee970b00-4e92-11eb-93d8-2d000dc73591.png">

This would let you use this plugin for both discord accounts without having to reconfigure it when you switch

**Comment by moonpost** [comment, marked as answer]
Having exact same problem. Catalina Mac. latest discord and stream deck. only get AUTHENTICATION_FAILED when attempting to use. please advise.

**Reply by Distrouk** [reply]
Had same problem all you need to do is also enter the CLIENT SECRET on discord developers site in the app Secret on the stream deck app

**Reply by moonpost** [reply]
yeah, did that. several times. didn't work.  finally decided to REGENERATE the Client Secret and then was finally able to get it working.
